Zend/zend_alloc.c

```diff
@@ -2603,6 +2734,7 @@ ZEND_API size_t ZEND_FASTCALL _zend_mm_block_size(zend_mm_heap *heap, void *ptr
 
 typedef struct _zend_alloc_globals {
 	zend_mm_heap *mm_heap;
+	int use_input_zone;
```

Comment:
Why a signed type?

Reply:
That's not necessary indeed :+1:

Reply:
The linux kernel has saturating types, based on [PAX_REFCOUNT](https://forums.grsecurity.net/viewtopic.php?t=4173), to prevent wrap-around, but I think it's overkill here: an attacker shouldn't™ able to overflow a `size_t` or something is seriously wrong.